# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.h
# Commit: 909001ac06d8
# Full Hash: 909001ac06d8b3982116a0b0f753ae365b1cea22
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2020-07-23 03:20:25
# Regressor Bug: 1652270
# File Overlap Count: 1
# Description:
#   Bug 1652270. Convert nsFrameLoader::Print to return a Promise. r=farre,remote-protocol-reviewers,marionette-reviewers,jgraham,whimboo,mixedpuppy
#   
#   (Instead of requiring callers to pass an nsIWebProgressListener.)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83249
# ==============================================================================

diff -r 8cfbd5213d04 -r 909001ac06d8 dom/base/nsFrameLoader.h
--- a/dom/base/nsFrameLoader.h	Thu Jul 23 02:01:54 2020 +0300
+++ b/dom/base/nsFrameLoader.h	Wed Jul 22 23:17:45 2020 +0000
@@ -24,6 +24,7 @@
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/Element.h"
 #include "mozilla/dom/Nullable.h"
+#include "mozilla/dom/Promise.h"
 #include "mozilla/dom/WindowProxyHolder.h"
 #include "mozilla/layers/LayersTypes.h"
 #include "nsCOMPtr.h"
@@ -105,6 +106,7 @@
   typedef mozilla::dom::BrowserBridgeChild BrowserBridgeChild;
   typedef mozilla::dom::BrowsingContext BrowsingContext;
   typedef mozilla::dom::BrowsingContextGroup BrowsingContextGroup;
+  typedef mozilla::dom::Promise Promise;
 
  public:
   // Called by Frame Elements to create a new FrameLoader.
@@ -223,9 +225,9 @@
 
   void RequestSHistoryUpdate(bool aImmediately = false);
 
-  void Print(uint64_t aOuterWindowID, nsIPrintSettings* aPrintSettings,
-             nsIWebProgressListener* aProgressListener,
-             mozilla::ErrorResult& aRv);
+  already_AddRefed<Promise> Print(uint64_t aOuterWindowID,
+                                  nsIPrintSettings* aPrintSettings,
+                                  mozilla::ErrorResult& aRv);
 
   void StartPersistence(BrowsingContext* aContext,
                         nsIWebBrowserPersistDocumentReceiver* aRecv,